packages/eslint-plugin/src/stylex-sort-keys.js

```diff
@@ -0,0 +1,291 @@
+import getPropertyPriority from './utils/getPropertyPriority';
+/*:: import { Rule } from 'eslint'; */
+
+type Schema = {
+  validImports: Array<string>,
+  minKeys: number,
+  allowLineSeparatedGroups: boolean,
+};
+
+type Stack = null | {
+  upper: Stack,
+  prevNode: Property | null,
+  prevName: string | null,
+  prevBlankLine: boolean,
+  numKeys: number,
+};
+
+function isAtRule(key: string) {
+  return key.startsWith('@');
+}
+
+function isPseudoClassOrElement(key: string) {
+  return key.startsWith(':');
+}
+
+function isValidOrder(prevName: string, currName: string): boolean {
+  if (
+    isAtRule(prevName) ||
+    isPseudoClassOrElement(prevName) ||
+    isAtRule(currName) || isPseudoClassOrElement(currName)
+  ) {
+    return getPropertyPriority(prevName) <= getPropertyPriority(currName);
+  }
+
+  return prevName <= currName;
+}
+
+function getPropertyName(node: Property): string | null {
+  const staticName = getStaticPropertyName(node);
+
+  return staticName !== null ? staticName : node.key.name || null;
+}
+
+const stylexSortKeys = {
+  type: 'suggestion',
+  docs: {
+    description: 'Require style properties to be sorted by key',
+    recommended: false,
+    url: 'https://github.com/facebook/stylex/tree/main/packages/eslint-plugin',
+  },
+  schema: [
+    {
+      type: 'object',
+      properties: {
+        validImports: {
+          type: 'array',
+          items: { type: 'string' },
+          default: ['stylex', '@stylexjs/stylex'],
+        },
+        minKeys: {
+          type: 'integer',
+          minimum: 2,
+          default: 2,
+        },
+        allowLineSeparatedGroups: {
+          type: 'boolean',
+          default: false,
+        },
+      },
+      additionalProperties: false,
+    },
+  ],
+  create(context: Rule.RuleContext): { ... } {
+    const {
+      validImports: importsToLookFor = ['stylex', '@stylexjs/stylex'],
+      minKeys = 2,
+      allowLineSeparatedGroups = false,
+    }: Schema = context.options[0] || {};
+
+    const styleXDefaultImports = new Set<string>();
+    const styleXCreateImports = new Set<string>();
+
+    function isStylexCallee(node: Node) {
+      return (
+        (node.type === 'MemberExpression' &&
+          node.object.type === 'Identifier' &&
+          styleXDefaultImports.has(node.object.name) &&
+          node.property.type === 'Identifier' &&
+          node.property.name === 'create') ||
+        (node.type === 'Identifier' && styleXCreateImports.has(node.name))
+      );
+    }
+
+    function isStylexDeclaration(node: $ReadOnly<{ ...Node, ... }>) {
+      return (
+        node &&
+        node.type === 'CallExpression' &&
+        isStylexCallee(node.callee) &&
+        node.arguments.length === 1
+      );
+    }
+
+    let stack: Stack = null;
+    let isInsideStyleXCreateCall = false;
+    let objectExpressionNestingLevel = -1;
+
+    return {
+      ImportDeclaration(node: ImportDeclaration) {
```

Comment:
just realized that we should do the same with `require`

Reply:
It might be OK to skip that. `require` is not fully supported. Theming only works with ESM for example.